### PR TITLE
Add latency metrics for etcd operations

### PR DESCRIPTION
**Pull request by fgrzadkowski**
This adds some metrics listed #1625 

@davidopp @wojtek-t

### CODE REVIEW

File: pkg/tools/etcd_helper.go [hunk 1]
```diff
@@ -55,12 +56,32 @@ var (
...
+	etcdRequestLatenciesSummary = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "etcd_request_latencies_summary",
+			Help: "Etcd request latency summary in microseconds for each operation and object type.",
+		},
+		[]string{"operation", "type"},
+	)
 )
 
+func getTypeNamefromObject(obj runtime.Object) string {
```
Comment (wojtek-t) [inline pkg/tools/etcd_helper.go]: s/from/From/

same below

File: pkg/tools/etcd_helper.go [hunk 2]
```diff
@@ -55,12 +56,32 @@ var (
...
+	return reflect.TypeOf(obj).String()
+}
+
+func getTypeNamefromObjectSlice(objSlice interface{}) string {
+	return reflect.TypeOf(objSlice).String()
```
Comment (wojtek-t) [inline pkg/tools/etcd_helper.go]: The implementation here is exactly the same as for getTypeNameFromObject. So why do we need both?
Or maybe it should be "reflect.TypeOf(objSlice).Elem().String()"?

**Comment by mbforbes** [comment]
Assigning to @davidopp; didn't rope in many others from #1625 as this is a relatively small PR.

**Comment by a-robinson** [comment]
/sub - I'm interested in at least being notified of new metrics being added to the master process

File: pkg/tools/etcd_helper.go (hunk 2 above)
Comment (fgrzadkowski) [inline pkg/tools/etcd_helper.go]: Thanks for catching this :) I was experimenting with different implementations and ended up where they are identical.

**Comment by davidopp** [comment]
LGTM

**Pull request merged**